gprofiler/spark/spark_collector.py

```diff
@@ -41,6 +41,8 @@
 from gprofiler.utils.fs import escape_filename
 from gprofiler.utils.process import search_for_process
 
+MAX_FIND_ATTEMPTS = 10
```

Comment:
We want 10 minutes but the interval might be e.g 30s in which case max attempts is 20.

Instead - get the monotonic time when starting and try up to 10 minutes?